components/Files/UploadAndList.tsx

```diff
@@ -62,8 +101,22 @@ export default function UploadAndList() {
                     {f.file_name}
                   </div>
                 </div>
-              </div>
-            ))}
+              );
+
+              return f.is_directory ? (
+                <Link
+                  key={f.id}
+                  href={`?path=${encodeURIComponent(targetPath!)}`}
+                  className="rounded-md p-2 text-sm hover:bg-accent/40 block"
+                >
+                  {TileContent}
+                </Link>
+              ) : (
+                <div key={f.id} className="rounded-md p-2 text-sm hover:bg-accent/40">
+                  {TileContent}
+                </div>
+              );
+            })}
```

Comment:
Max file length
- actual: file exceeds 100 lines of code.
- required: no file is longer than 100 lines of code.